tests/suites/test_suite_cipher.chachapoly.data

```diff
@@ -1,6 +1,6 @@
 Decrypt empty buffer
 depends_on:MBEDTLS_CHACHAPOLY_C
-dec_empty_buf:
+dec_empty_buf:MBEDTLS_CIPHER_CHACHA20_POLY1305
```

Comment:
That's an authenticated cipher, so it should return `MBEDTLS_ERR_CIPHER_AUTH_FAILED`.

Reply:
mbedtls_cipher_update_ad() may return this, but finish doesn't as currently implemented (which may be a bug).

Reply:
It wouldn't make sense for `mbedtls_cipher_update_ad` to return `AUTH_FAILED`. OTOH `finish` doesn't make sense for an authenticated cipher — it should return something like `BAD_INPUT_DATA` or `INVALID_CONTEXT`.

Reply:
Ah, no, sorry, that's not how the API works. `finish` should return 0 on empty input, and _then_ the application should call `mbedtls_cipher_check_tag`.